### PR TITLE
restart function for band structures

### PR DESCRIPTION
**Pull request by FeLoch**
new function for creating restart job for band structure calculations.

### CODE REVIEW

**Comment by jan-janssen** [comment]
I am a bit confused. I guess the idea was that both `restart_for_band_structure_calculations` should have exactly the same input parameters, just like `calc_md`. If that is the case, then we should also add a function template in https://github.com/pyiron/pyiron/blob/master/pyiron/dft/job/generic.py

**Comment by FeLoch** [comment]
Should now be done!

**Comment by jan-janssen** [comment]
Can you also add a function template to generic DFT ? 
https://github.com/pyiron/pyiron/blob/master/pyiron/dft/job/generic.py#L130

**Comment by FeLoch** [comment]
> 
> 
> Can you also add a function template to generic DFT ?
> https://github.com/pyiron/pyiron/blob/master/pyiron/dft/job/generic.py#L130

done

File: pyiron/dft/job/generic.py [hunk 1]
```diff
@@ -143,6 +143,11 @@ def set_mixing_parameters(
     #    self.set_kpoints(mesh=mesh, scheme='MP', center_shift=None, symmetry_reduction=True, manual_kpoints=None,
     #                     weights=None, reciprocal=True)
 
+    def restart_for_band_structure_calculations(self, job_name=None):
```
Comment (jan-janssen) [inline pyiron/dft/job/generic.py:146]: Can you copy the Docstring as well?

Comment (FeLoch) [reply pyiron/dft/job/generic.py:146]: done

**Comment by jan-janssen** [comment]
Looks good to me.

**Pull request merged**